src/algebra/Module/basic.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright (c) 2019 Robert A. Spencer. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Robert A. Spencer
+-/
+import algebra.module
+import algebra.punit_instances
+import category_theory.types
+import linear_algebra.basic
+open category_theory
+
+universe u
+
+variables (R : Type u) [ring R]
+
+/-- An example of a module for any ring is the zero module -/
+instance module.zero_module : module R punit := module.of_core $
+  by refine {
+    smul := λ _ _, punit.star,
+    .. punit.comm_ring, ..};
+    intros; exact subsingleton.elim _ _
+
+/-- The category of R-modules and their morphisms. -/
+structure Module :=
+  (carrier : Type)
```

Comment:
```suggestion
  (carrier : Type v)
```
where `v` is some new universe variable.

Reply:
Oops, missed that! Absolutely.

Reply:
So if I do (as I think you are intending)

```lean
universes u v

variables (R : Type u) [ring R]

/-- The category of R-modules and their morphisms. -/
structure Module :=
  (carrier : Type v)
  [is_add_comm_group : add_comm_group carrier]
  [is_module : module R carrier]

...

instance : category (Module R) := {
  hom := λ M N, M →ₗ[R] N,
  id := λ M, 1 ,
  comp := λ A B C f g, g.comp f ,
}
```
Then the instance line complains with
```
invalid universe declaration, 'u_1' shadows a local universe
```
I'm not quite sure where it is inserting the universe `u_1`: is there a quick way to check?

Reply:
Oh, sorry, I missed that Johan was suggesting putting in an independent universe level. Let's change `(carrier : Type v)` to `(carrier : Type u)`. I've never heard of anyone worrying that their modules and their rings live in different universes.

Hopefully this will make this issue go away.

Reply:
I went ahead and resolved some conversations that were clearly done, but I will leave this one :)